doc/dev/references/index.md

```diff
@@ -0,0 +1,32 @@
+Reference docs for Juju developers
+
+- [Agent](agent.md)
+- [Agent introspection](https://discourse.charmhub.io/t/117)
```

Comment:
The agent introspection titles should be removed -- that's content we're moving to the user docs (as it's actually user content, for advanced troubleshooting).